sys/include/net/conn.h

```diff
@@ -73,27 +75,16 @@
 #ifndef NET_CONN_H_
 #define NET_CONN_H_
 
+#include <stdbool.h>
+
 #include "net/conn/ip.h"
 #include "net/conn/tcp.h"
 #include "net/conn/udp.h"
-#include "net/ipv6/addr.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-/**
- * @brief   Find the best matching source address for a given prefix
- *
- * @param[in] dst   Pointer to the IPv6 address to find a match for
- *                  Must not be NULL
- *
- * @return NULL if no matching address on any interface could be found
- * @return pointer to an IPv6 address configured on an interface with the best
- *         match to @p dst
- */
-ipv6_addr_t *conn_find_best_source(const ipv6_addr_t *dst);
```

Comment:
As stated in the description: the new implicit binding capabilities of the API make this function unnecessary.